crates/shielded_token/src/lib.rs

```diff
@@ -42,6 +45,56 @@ pub use storage::*;
 #[cfg(feature = "masp")]
 pub use crate::masp::shielded_wallet::ShieldedWallet;
 
+/// An object with multiple representations
+#[derive(
+    Clone,
+    Copy,
+    Debug,
+    PartialEq,
+    Eq,
+    PartialOrd,
+    Ord,
+    Hash,
+    BorshSerialize,
+    BorshDeserialize,
+    BorshSchema,
+    Deserialize,
+    Serialize,
+)]
+// This object can be deserialized from either form
+#[serde(untagged)]
+pub enum Poly<X, Y> {
+    /// The primary way this object can be represented
+    Left(X),
+    /// The secondry way this object can be represented
+    Right(Y),
+}
```

Comment:
this seems unnecessary, the `either` crate is imported into the workspace. perhaps the only problem is not having borsh serialization code